android/app/src/main/java/ru/spbstu/amd/learnbraille/screens/practice/PracticeFragment.kt

```diff
@@ -82,12 +105,31 @@ class PracticeFragment : Fragment() {
             viewModel.onCorrectComplete()
         })
 
+        hintButton.setOnClickListener {
+            viewModel.onHint()
+            Toast.makeText(context, viewModel.getDotsString(), Toast.LENGTH_SHORT).show()
+            Timber.i("Hint invoked")
+            val expectedDots = viewModel.getExpectedDots()
+            Timber.i(expectedDots.toString())
+            dotCheckBoxes[0].isChecked = expectedDots?.b1 == BrailleDot.F
+            dotCheckBoxes[1].isChecked = expectedDots?.b2 == BrailleDot.F
+            dotCheckBoxes[2].isChecked = expectedDots?.b3 == BrailleDot.F
+            dotCheckBoxes[3].isChecked = expectedDots?.b4 == BrailleDot.F
+            dotCheckBoxes[4].isChecked = expectedDots?.b5 == BrailleDot.F
+            dotCheckBoxes[5].isChecked = expectedDots?.b6 == BrailleDot.F
+            dotCheckBoxes.forEach { it.isClickable = false }
+
+            if (expectedDots != null) {
+                serial.trySend(expectedDots)
+            }
+        }
+
         viewModel.eventIncorrect.observe(this@PracticeFragment, Observer {
             if (!it) {
                 return@Observer
             }
 
-            Toast.makeText(context, "Incorrect!", Toast.LENGTH_SHORT).show()
+            Toast.makeText(context, "Неправильно!", Toast.LENGTH_SHORT).show()

```

Comment:
UI string should be in the resource file.

Reply:
ok, extracted